boards/riscv32/hifive1/clock.c

```diff
@@ -0,0 +1,29 @@
+/*
+ * Copyright (c) 2017 Jean-Paul Etienne <fractalclone@gmail.com>
+ * Copyright (c) 2017 Palmer Dabbelt <palmer@dabbelt.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <init.h>
+#include <board.h>
+#include "prci.h"
+
+/* Selects the 16MHz oscilator on the HiFive1 board, which provides a clock
+ * that's accurate enough to actually drive serial ports off of.
+ */
+static int hifive1_clock_init(struct device *dev)
+{
+	ARG_UNUSED(dev);
+	u32_t config_value = 0;
+
+	config_value |= PLL_REFSEL(1);
+	config_value |= PLL_BYPASS(1);
+	PRCI_REG(PRCI_PLLCFG) = config_value;
```

Comment:
Any reason to not just have:

PRCI_REG(PRCI_PLLCFG) = PLL_REFSEL(1) | PLL_BYPASS(1);